Meadowlark-js/packages/meadowlark-core/src/handler/Delete.ts

```diff
@@ -21,58 +24,63 @@ const moduleName = 'Delete';
  *
  * Validates resource and forwards to backend for deletion
  */
-export async function deleteIt(event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> {
-  const { awsRequestId } = context;
+export async function deleteIt(frontendRequest: FrontendRequest): Promise<FrontendResponse> {
   try {
-    writeRequestToLog(moduleName, event, context, 'deleteIt');
+    writeRequestToLog(moduleName, frontendRequest, 'deleteIt');
 
-    const { jwtStatus, errorResponse } = validateJwt(authorizationHeader(event));
+    const { jwtStatus, errorResponse } = validateJwt(authorizationHeader(frontendRequest));
     if (errorResponse != null) {
-      writeDebugStatusToLog(moduleName, context, 'deleteIt', errorResponse.statusCode, JSON.stringify(jwtStatus));
-      return errorResponse as APIGatewayProxyResult;
+      writeDebugStatusToLog(moduleName, frontendRequest, 'deleteIt', errorResponse.statusCode, JSON.stringify(jwtStatus));
+      return errorResponse as FrontendResponse;
     }
 
-    const pathComponents: PathComponents | null = pathComponentsFrom(event.path);
+    const pathComponents: PathComponents | null = pathComponentsFrom(frontendRequest.path);
     if (pathComponents === null || pathComponents.resourceId == null) {
-      writeDebugStatusToLog(moduleName, context, 'deleteIt', 404);
+      writeDebugStatusToLog(moduleName, frontendRequest, 'deleteIt', 404);
       return { body: '', statusCode: 404 };
     }
 
     const { documentInfo, errorBody, headerMetadata } = await validateResource(pathComponents, null);
     if (errorBody !== null) {
       const statusCode = documentInfo === NoDocumentInfo ? 404 : 400;
-      writeDebugStatusToLog(moduleName, context, 'deleteIt', statusCode, errorBody);
+      writeDebugStatusToLog(moduleName, frontendRequest, 'deleteIt', statusCode, errorBody);
       return { body: errorBody, statusCode, headers: headerMetadata };
     }
 
-    const { result, failureMessage } = await getBackendPlugin().deleteDocumentById({
+    const request: DeleteRequest = {
       id: pathComponents.resourceId,
       documentInfo,
-      validate: event.headers['reference-validation'] !== 'false',
+      validate: frontendRequest.headers['reference-validation'] !== 'false',
       security: {
         ...newSecurity(),
         isOwnershipEnabled: jwtStatus.isOwnershipEnabled,
         clientName: jwtStatus.subject,
       },
-      traceId: awsRequestId,
-    });
+      traceId: frontendRequest.traceId,
+    };
 
-    if (result === 'DELETE_SUCCESS') {
-      writeDebugStatusToLog(moduleName, context, 'deleteIt', 204);
+    beforeDeleteDocumentById(request);
```

Comment:
Example "before" event publish, here before delete attempt in the document store